app/helpers/warning_bar_helper.rb

```diff
@@ -47,6 +53,17 @@ def setting_hostname_mismatched?
     Setting.host_name.gsub(/:\d+$/, '') != request.host
   end
 
+  def no_workflow_for_wp_edit_role?
+    workflow_exists = OpenProject::Cache.read('no_wp_share_editor_workflow')
+
+    if workflow_exists.nil?
+      workflow_exists = Workflow.exists?(role_id: Role.where(builtin: Role::BUILTIN_WORK_PACKAGE_EDITOR).select(:id))
+      OpenProject::Cache.write('no_wp_share_editor_workflow', workflow_exists) if workflow_exists
+    end
+
+    !workflow_exists
+  end
```

Comment:
I don't see that the cache is ever cleaned? So wouldn't the warning always persist even after the workflow has been created?

Reply:
If I wrote it correctly, and there might be an easier way of doing this, the cache is only written if workflows exist:

`OpenProject::Cache.write(...) if workflow_exists`

That is why I did not use the `Cache.fetch` method which would be way simpler to use.

Reply:
👍